### PR TITLE
feat: send js-ceramic and ceramic-one versions to CAS when creating requests

### PR DESCRIPTION
**Pull request by stbrody**
js-ceramic half corresponding to https://github.com/ceramicnetwork/ceramic-anchor-service/pull/1232

Builds on https://github.com/ceramicnetwork/js-ceramic/pull/3245

### CODE REVIEW

File: packages/core/src/ceramic.ts [hunk 1]
```diff
@@ -535,11 +547,10 @@ export class Ceramic implements StreamReaderWriter, StreamStateLoader {
   }
 
   async _publishVersionMetrics() {
-    const ipfsVersion = (await this.ipfs.id()).agentVersion
     Metrics.observe(VERSION_INFO, 1, {
       jsCeramicVersion: this._versionInfo.cliPackageVersion,
       jsCeramicGitHash: this._versionInfo.gitHash,
-      ceramicOneVersion: ipfsVersion,
+      ceramicOneVersion: this._versionInfo.ceramicOneVersion,
```
Comment (stbrody) [inline packages/core/src/ceramic.ts:553]: a small downside of this is that if ipfs/ceramic-one is upgraded out from under a running js-ceramic node, the js-ceramic node will continue to report the old version of ipfs/C1 until js-ceramic restarts.  Since it should be rare to upgrade C1 without also at least restarting js-ceramic, I think this is acceptable.

File: packages/cli/src/__tests__/ceramic-multi-daemon.test.ts [hunk 2]
```diff
@@ -16,16 +16,20 @@ const seed = 'SEED'
...
-      db: `sqlite://${stateStoreDirectory}/ceramic.sqlite`,
-      disableComposedb: false,
+  const core = await Ceramic.create(
+    ipfs,
+    {
+      pubsubTopic: TOPIC,
+      stateStoreDirectory,
+      anchorOnRequest: false,
+      indexing: {
+        db: `sqlite://${stateStoreDirectory}/ceramic.sqlite`,
```
Comment (gvelez17) [inline packages/cli/src/__tests__/ceramic-multi-daemon.test.ts:26]: is sqlite hardcoded in?  just a side question

File: packages/core/src/__tests__/ceramic-pinning.test.ts [hunk 3]
```diff
@@ -47,18 +47,22 @@ const createCeramic = async (
...
+    {
+      stateStoreDirectory,
+      anchorOnRequest,
+      indexing: {
+        db: connectionString.href,
+        allowQueriesBeforeHistoricalSync: true,
+        disableComposedb: true,
+        enableHistoricalSync: false,
+      },
+      pubsubTopic: '/ceramic/inmemory/test', // necessary so Ceramic instances can talk to each other
```
Comment (gvelez17) [inline packages/core/src/__tests__/ceramic-pinning.test.ts:61]: would like to see a constant for this

File: packages/cli/src/__tests__/ceramic-multi-daemon.test.ts (hunk 2 above)
Comment (stbrody) [reply packages/cli/src/__tests__/ceramic-multi-daemon.test.ts:26]: This is a test file, so yes this test always uses sqlite

File: packages/core/src/ceramic.ts (hunk 1 above)
Comment (gvelez17) [reply packages/core/src/ceramic.ts:553]: yeah i think thats a really weird edge case, and we can also tell when it happens.  i think this metric is fine

File: packages/core/src/__tests__/ceramic-pinning.test.ts (hunk 3 above)
Comment (stbrody) [reply packages/core/src/__tests__/ceramic-pinning.test.ts:61]: this is just a test file, the value used here could be anything

**Review by gvelez17 — APPROVED** [review]
had a small change request but not critical, approved

**Review by smrz2001 — APPROVED** [review]
:shipit:
